ext/ffi/jit_trampoline.rs

```diff
@@ -83,20 +96,35 @@ pub(crate) fn codegen(sym: &crate::Symbol) -> String {
     c += native_arg_to_c(ty);
     let _ = write!(c, " p{i}");
   }
+  if needs_unwrap {
+    let _ = write!(c, ", struct FastApiTypedArray* p_ret");
+  }
   c += ") {\n";
-  // return func(p0, p1, ...);
-  c += "  return func(";
-  for (i, ty) in sym.parameter_types.iter().enumerate() {
-    if i > 0 {
-      c += ", ";
-    }
-    if matches!(ty, NativeType::Pointer) {
-      let _ = write!(c, "p{i}->data");
-    } else {
-      let _ = write!(c, "p{i}");
+  // func(p0, p1, ...);
+  let mut call_s = String::from("func(");
+  {
+    for (i, ty) in sym.parameter_types.iter().enumerate() {
+      if i > 0 {
+        call_s += ", ";
+      }
+      if matches!(ty, NativeType::Pointer) {
+        let _ = write!(call_s, "p{i}->data");
+      } else {
+        let _ = write!(call_s, "p{i}");
+      }
     }
+    call_s += ");\n";
   }
-  c += ");\n}\n\n";
+  if needs_unwrap {
+    // <return_type> r = func(p0, p1, ...);
+    let ret = native_to_c(&sym.result_type);
+    let _ = write!(c, " {ret} r = {call_s}");
+    c += " p_ret->data = (uint32_t *)&r;\n";
```

Comment:
I'm confused. You're taking a pointer to something on the stack - this seems dangerous. Why not copy the value into p_ret?

Reply:
Yeah, I was starting to look at this as well, this seems wrong. Should this be something like this instead:

```c
p_ret->data[0] = r;
```
?

Reply:
fixed